File: TransCoder_tokenized_test_set_functions/python/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.py
def maximumPalinUsingKChanges ( strr, k ) :
    palin = strr
    l = 0
    r = len ( strr ) - 1
    while ( l <= r ) :
        if ( strr [ l ] != strr [ r ] ) :
            palin [ l ] = palin [ r ] = max ( strr [ l ], strr [ r ] )
            k -= 1
        l += 1
        r -= 1
    if ( k < 0 ) :
        return "Not possible"
    l = 0
    r = len ( strr ) - 1
    while ( l <= r ) :
        if ( l == r ) :
            if ( k > 0 ) :
                palin [ l ] = '9'
        if ( palin [ l ] < '9' ) :
            if ( k >= 2 and palin [ l ] == strr [ l ] and palin [ r ] == strr [ r ] ) :
                k -= 1
                palin [ l ] = palin [ r ] = '9'
            elif ( k >= 1 and ( palin [ l ] != strr [ l ] or palin [ r ] != strr [ r ] ) ) :
                k -= 1
                palin [ l ] = palin [ r ] = '9'
        l += 1
        r -= 1
    return palin

File: results/java/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.cpp
string f_filled ( string str , int k ) {
  string palin ;
  string ans ;
  int l = 0 ;
  int r = str . length ( ) - 1 ;
  while ( l < r ) {
    if ( str [ l ] != str [ r ] ) {
      palin [ l ] = palin [ r ] = ( char ) max ( str [ l ] , str [ r ] ) ;
      k -- ;
    }
    l ++ ;
    r -- ;
  }
  if ( k < 0 ) return "Not possible" ;
  l = 0 ;
  r = str . length ( ) - 1 ;
  while ( l <= r ) {
    if ( l == r ) {
      if ( k > 0 ) palin [ l ] = '9' ;
    }
    if ( palin [ l ] < '9' ) {
      if ( k >= 2 && palin [ l ] == str [ l ] && palin [ r ] == str [ r ] ) {
        k -= 2 ;
        palin [ l ] = palin [ r ] = '9' ;
      }
      else if ( k >= 1 && ( palin [ l ] != str [ l ] || palin [ r ] != str [ r ] ) ) {
        k -- ;
        palin [ l ] = palin [ r ] = '9' ;
      }
    }
    l ++ ;
    r -- ;
  }
  for ( int i = 0 ;
  i < palin . length ( ) ;
  i ++ ) {
    ans += palin [ i ] ;
  }
  return ans ;
}
